src/exception.c

```diff
@@ -16,6 +16,11 @@
 #include <stdbool.h>
 #include <math.h>
 
+/** Invalid TLS Minimum Address */
+#define TLS_INVALID_MIN (uint32_t)(KERNEL_TP_INVALID-28672)
+/** TLS Size */
+#define TLS_SIZE ((uint32_t)(__tls_end)-(uint32_t)(__tls_base))
```

Comment:
Can you maybe move these to kernel_internal.h? You could move there also TP_OFFSET so that you reuse the magic offset.